tc_aws/aws/bucket.py

```diff
@@ -34,23 +42,20 @@ def __init__(self, bucket, region, endpoint, max_retry=None):
         :return: The created bucket
         """
         self._bucket = bucket
-
-        session = aiobotocore.get_session()
-
         config = None
         if max_retry is not None:
             config = Config(
                 retries=dict(
                     max_attempts=max_retry
                 )
             )
-
-        self._client = session.create_client(
-            's3',
-            region_name=region,
-            endpoint_url=endpoint,
-            config=config
-        )
+        if self._client is None:
```

Comment:
so, from what I understand, this is the only required change here. Other changes are rather cosmetic.

Reply:
Yes, this is the operative change.  The rest are PEP8 changes and very minor performance changes.